src/store/user/user-context.js

```diff
@@ -10,6 +11,27 @@ export const UserContextProvider = ({ children }) => {
   const [showMemberRoleUpdateModal, setShowMemberRoleUpdateModal] =
     useState(false);
   const [selectedMember, setSelectedMember] = useState(null);
+  const [superUserVerified, setSuperUserVerified] = useState(false);
+
+  const verifySuperUser = async () => {
+    let selfUser = null;
+    let selfIsSuperUser = false;
+    let selfSuperUserVerified = false;
+    try {
+      const { data } = await getUserSelf();
```

Comment:
If the API fails/errors, will we never set `selfSuperUserVerified`, potentially causing infinite loops in consuming components e.g [here](https://github.com/Real-Dev-Squad/website-members/pull/410/files#r953219362)?

Reply:
even if the API fails/errors after the try-catch block we call `setUserApiCalled` which will set it to `false`( the initial value we initialized `selfUserApiCalled` with).